controllers/medusa/medusabackupjob_controller.go

```diff
@@ -186,6 +190,11 @@ func (r *MedusaBackupJobReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			logger.Error(err, "Failed to get backup summary")
 			return ctrl.Result{}, err
 		}
+		if backupSummary == nil {
```

Comment:
suggestion: Would the same approach be attainable, if we in the getBackupSummary instead of "nil, nil" (sounding like it's an acceptable situation) instead returned nil, errors.TerminalError(errors.New("backup summary couldn't be found")) ?

That prevents requeue also, but it also indicates in the function itself that the "nil, nil" is not an okay result.

Reply:
Yes, I think this would be a better solution. I had to tweak the error lib imports to get this somewhat pretty, but I still can't import `TerminalError`. Do we need it? Which library do I find it in?

Reply:
It's in "sigs.k8s.io/controller-runtime/pkg/reconcile"

reconcile.TerminalError(err)

Reply:
Thanks, I've updated what we return.